Il2CppInterop.Runtime/Injection/Hooks/GarbageCollector_RunFinalizer_Patch.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Il2CppInterop.Runtime.Runtime;
+
+namespace Il2CppInterop.Runtime.Injection.Hooks;
+
+internal class GarbageCollector_RunFinalizer_Patch : Hook<GarbageCollector_RunFinalizer_Patch.MethodDelegate>
+{
+    public override string TargetMethodName => "GarbageCollector::RunFinalizer";
```

Comment:
What happens to classes without a finalizer? Looking at libil2cpp looks like this is called only for classes that have it.
Feels a bit hacky but we could try calling `Il2CppSystem.GC.ReRegisterForFinalize` for them when adding to s_cache and then skip the original `GarbageCollector::RunFinalizer`